src/core/public/notifications/toasts/error_toast.tsx

```diff
@@ -31,18 +31,21 @@ import {
 } from '@elastic/eui';
 import { EuiSpacer } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
-
-import { OverlayStart } from '../../overlays';
+import { OverlayStart } from 'kibana/public';
 import { I18nStart } from '../../i18n';
 
 interface ErrorToastProps {
   title: string;
-  error: Error;
+  error: RequestError;
```

Comment:
I would:
1/ keep `Error` in `ErrorToastProps` as RequestError does not reflect the real type (is can be any kind of error)
2/ add a typeguard for `RequestError`: `isRequestError = (e: Error): e is RequestError => error.body?.attributes?.error?.caused_by !== undefined`

Use it in the block
```ts
if (isRequestError(error)) {
    text += `${error.body.attributes.error.caused_by.type}\n`;
    text += `${error.body.attributes.error.caused_by.reason}\n\n`;
  }
```

Reply:
thx! adapted the code in this way

Reply:
dear @pgayvallet , have some TypeScript errors here to resolve 

```Typescript
const isRequestError = (e: Error): e is RequestError => e.body?.attributes?.error?.caused_by !== undefined;
```

Error message: `Property 'body' does not exist on type 'Error'.`

How can I resolve this? many thx

Reply:
`(e: Error)` -> `(e: any)` should do the trick

Reply:
thx, to omit any, I've implemented it the following way:

```TypeScript
const isRequestError = (e: Error | RequestError): e is RequestError => {
   if ('body' in e) {
     return e.body?.attributes?.error?.caused_by !== undefined;
   }
   return false;
 };
```